securebanking-openbanking-uk-gateway-conversion-filter/src/main/java/com/forgerock/securebanking/uk/gateway/conversion/filter/IntentConverterFilter.java

```diff
@@ -0,0 +1,239 @@
+import org.forgerock.util.promise.NeverThrowsException;
+import org.forgerock.util.promise.Promise;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.net.URI;
+
+import static com.forgerock.securebanking.uk.gateway.utils.IntentTypeUtils.getIntentType;
+import static org.forgerock.http.protocol.Entity.APPLICATION_JSON_CHARSET_UTF_8;
+import static org.forgerock.json.JsonValueFunctions.enumConstant;
+import static org.forgerock.util.promise.Promises.newResultPromise;
+
+/**
+ * Filter to convert IDM json intent objects to OB data model objects.
+ *
+ * This filter must have received {@link  MessageType} as required to get the entity  to instance the converter and know the {@link MessageImpl} to be updated<br/>
+ * Failures from the `Converter instance` is either a {@link RuntimeException} that will catch to build a {@link ResponseException}.
+ *
+ * Configuration options:
+ *
+ * <pre>
+ * {@code {
+ *      "name": "IntentConverterFilter-name"
+ *      "type": "IntentConverterFilter",
+ *      "config": {
+ *         "messageType"  MessageType   [REQUIRED - The type of message for which to convert the entity, Must be either "REQUEST" or "RESPONSE"]
+ *      }
+ *  }
+ *  }
+ * </pre>
+ * <p>Examples</p>
+ * <pre>
+ * {@code {
+ *      "name": "IntentConverterFilter-name"
+ *      "type": "IntentConverterFilter",
+ *      "config": {
+ *         "messageType: "request"
+ *      }
+ *  }
+ *  }
+ * </pre>
+ * <pre>
+ *  {@code {
+ *       "name": "IntentConverterFilter-name"
+ *       "type": "IntentConverterFilter",
+ *       "config": {
+ *          "messageType: "response"
+ *       }
+ *   }
+ *   }
+ *  </pre>
+ */
+public class IntentConverterFilter implements Filter {
+
+    private static final Logger logger = LoggerFactory.getLogger(IntentConverterFilter.class);
+
+    private final MessageType messageType;
+    private static final ObjectMapper MAPPER = GenericConverterMapper.getMapper();
+
+    /**
+     * Constructor
+     * @param messageType
+     */
+    public IntentConverterFilter(final MessageType messageType) {
+        this.messageType = messageType;
+    }
+
+    @Override
+    public Promise<Response, NeverThrowsException> filter(Context context, Request request, Handler next) {
+        if (logger.isInfoEnabled()) {
+            printFilterInfo();
+        }
+
+        // MessageType.REQUEST
+        if (messageType.equals(MessageType.REQUEST)) {
+            try {
+                process(request, request.getUri().asURI());
+                return next.handle(context, request);
+            } catch (Exception e) {
+                return newResultPromise(badRequestError(e));
+            }
+        }
+
+        // MessageType.RESPONSE
+        return next.handle(context, request)
+                .then(response ->
+                        process(response, request.getUri().asURI())
+                )
+                .thenCatch(e ->
+                        badRequestError(e)
+                );
```

Comment:
I think this can be deleted, as the process method catches Exceptions and next.handle is tagged with NeverThrowsException

Reply:
Deleted.